src/main/java/se/liu/ida/hefquin/engine/queryproc/impl/optimizer/evolutionaryAlgorithm/InitializeNrGeneration.java

```diff
@@ -0,0 +1,39 @@
+package se.liu.ida.hefquin.engine.queryproc.impl.optimizer.evolutionaryAlgorithm;
+
+import se.liu.ida.hefquin.engine.queryplan.LogicalOperator;
+import se.liu.ida.hefquin.engine.queryplan.LogicalPlan;
+import se.liu.ida.hefquin.engine.queryplan.logical.impl.*;
+
+public class InitializeNrGeneration {
+
+    public static int countNumOfOp( final LogicalPlan plan ) {
+        final LogicalOperator lop = plan.getRootOperator();
+
+        int nrGeneration = 1;
+        if ( lop instanceof LogicalOpMultiwayJoin) {
+            int count = plan.numberOfSubPlans();
+            nrGeneration = calcFactorial(count);
+            for ( int i = 0; i < count; i++ ) {
+                nrGeneration = nrGeneration * countNumOfOp( plan.getSubPlan( i ) );
+            }
+        }
+        else if ( lop instanceof LogicalOpMultiwayUnion) {
+            int count = plan.numberOfSubPlans();
+            for ( int i = 0; i < count; i++ ) {
+                nrGeneration = nrGeneration * countNumOfOp( plan.getSubPlan( i ) );
+            }
+        }
+        else if ( lop instanceof LogicalOpRequest) {
+            nrGeneration = 1;
+        }
```

Comment:
In such cases, always add an else-branch to make sure we get to know when we have some unexpected input.
```suggestion
        }
        else {
            throw new IllegalArgumentException( lop.getClass().getName() );
        }
```